camus-etl-kafka/src/main/java/com/linkedin/camus/etl/kafka/CamusJob.java

```diff
@@ -476,9 +478,14 @@ public void sendTrackingCounts(JobContext job, FileSystem fs, Path newExecutionO
         }
       }
 
+      int publishCountMaxRetries =
+          Integer.parseInt(props.getProperty(ETL_PUBLISH_COUNT_MAX_RETRIES, ETL_PUBLISH_COUNT_MAX_RETRIES_DEFAULT));
       String brokerList = getKafkaBrokers(job);
       for (EtlCounts finalCounts : allCounts.values()) {
-        finalCounts.postTrackingCountToKafka(job.getConfiguration(), props.getProperty(KAFKA_MONITOR_TIER), brokerList);
+        for (int i = 0; i <= publishCountMaxRetries; i++) {
```

Comment:
Why do you still have that loop if the retry is handled inside ETLCounts already?